backend/src/crawler/udn_crawler.py

```diff
@@ -0,0 +1,148 @@
+    _fetch_news(self, page: int, search_term: str) -> list[Headline]: Helper method to fetch news headlines for a specific page.
+    _create_search_params(self, page: int, search_term: str): Creates the parameters for the search request.
+    _perform_request(self, params: dict): Performs the HTTP request to fetch news data.
+    _parse_headlines(response): Parses the response to extract headlines.
+    parse(self, url: str) -> News: Parses a news article from a given URL.
+    _extract_news(soup, url: str) -> News: Extracts news details from the BeautifulSoup object.
+    save(self, news: News, db: Session): Saves a news article to the database.
+    _commit_changes(db: Session): Commits the changes to the database with error handling.
+"""
+
+from requests import Response, get
+from bs4 import BeautifulSoup
+from sqlalchemy.orm import Session
+
+from .crawler_base import NewsCrawlerBase, Headline, News, NewsWithSummary
+from .exceptions import DomainMismatchException
+
+from ..models import NewsArticle
+
+class UDNNewsMetadata:
+    def __init__(self, page: int, id: str, channel_id: int, type: str) -> None:
+        self.page = page
+        self.id = id
+        self.channel_id = channel_id
+        self.type = type
+
+    @property
+    def to_dict(self) -> dict:
+        return dict(page=self.page, id=self.id, channelId=self.channel_id, type=self.type)
+
+class UDNCrawler(NewsCrawlerBase):
+    CHANNEL_ID = 2
+
+    def __init__(self, timeout: int = 5) -> None:
+        self.news_website_url = "https://udn.com/api/more"
+        self.news_website_news_child_urls = []
+        self.timeout = timeout
+
+    def startup(self, search_term: str) -> list[Headline]:
+        """
+        Initializes the application by fetching news headlines for a given search term across multiple pages.
+        This method is typically called at the beginning of the program when there is no data available,
+        hence it fetches headlines from the first 10 pages.
+
+        :param search_term: The term to search for in news headlines.
+        :return: A list of Headline namedtuples containing the title and URL of news articles.
+        :rtype: list[Headline]
+        """
+        return self.get_headline(search_term, page=(1, 10))
+
+    def get_headline(
+        self, search_term: str, page: int | tuple[int, int]
+    ) -> list[Headline]:
+
+        # Calculate the range of pages to fetch news from.
+        # If 'page' is a tuple, unpack it and create a range representing those pages (inclusive).
+        # If 'page' is an int, create a list containing only that single page number.
+        page_range = range(*page) if isinstance(page, tuple) else [page]
+        news_data = []
+        for page in page_range:
+            news_data.extend(self._fetch_news(page, search_term))
+
+        return news_data
+
+    def _fetch_news(self, page: int, search_term: str) -> list[Headline]:
```

Comment:
This method returns a list of `Headline` rather than a list of `News`, I'd suggestion renaming it to `_fetch_headlines`.
```suggestion
    def _fetch_headlines(self, page: int, search_term: str) -> list[Headline]:
```